agent/03_agent_deployment.sh

```diff
@@ -198,15 +198,47 @@ EOF
     set -x
 }
 
-function generate_fleeting_iso() {
-    export REPO_PATH=${WORKING_DIR}
+function sync_installer() {
```

Comment:
A possibility to avoid the duplication of `sync_repo_and_patch` could be to change https://github.com/openshift-metal3/dev-scripts/blob/894ff9dfa867fd9583557f020347d4800f9d2d64/utils.sh#L269 into 

```
DEFAULT_BRANCH=${DEFAULT_BRANCH:-$(git remote show origin | sed -n '/HEAD branch/s/.*: //p')}
```

And this would allow to override it in the `sync_installer()` function, ie: 

```
DEFAULT_BRANCH=agent-installer sync_repo_and_patch installer https://github.com/openshift/installer.git ${INSTALLER_PR}
```